File: go/resistor-color-duo/resistor_color_duo.go
package resistorcolorduo

var color = map[string]int{
	"black":0,
	"brown":1,
	"red":2,
	"orange":3,
	"yellow":4,
	"green":5,
	"blue":6,
	"violet":7,
	"grey":8,
	"white":9,
}

// Value should return the resistance value of a resistor with a given colors.
func Value(colors []string) int {
	
	var sum int

	for i,c := range colors {
			if i == 0 {
				sum += color[c]*10
			}else if i == 1{
				sum += color[c]
			}

		}
		return sum
	}
	

File: typescript/resistor-color-duo/resistor-color-duo.ts
export const COLORS = [
  'black',
  'brown',
  'red',
  'orange',
  'yellow',
  'green',
  'blue',
  'violet',
  'grey',
  'white',
];
export function decodedValue(colors: string[]):number {
  if(COLORS.indexOf(colors[1]) === 0) {
    return COLORS.indexOf(colors[0])*10;}
  return COLORS.indexOf(colors[0])*10+COLORS.indexOf(colors[1]);
}



